src/io.c

```diff
@@ -1,4478 +1,4641 @@
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *ip_int++ = (rng_state_t) P[pindex].rngStateAtBirth;
+                    n++;
+                }
+            break;
+
+        case IO_SLUG_STATE_INT:
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *ip_int++ = (uint64_t) P[pindex].id;
+                    *ip_int++ = (uint64_t) P[pindex].stoch_sn;
+                    n++;
+                }
+            break;
+
+        case IO_SLUG_STATE_DOUBLE: /*I did not include the last three quantities since I dont know how to deal with N */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type) /* Since only npart > 0 is saved, this line is not important */
+                {
+                    *fp++ = (double) P[pindex].targetMass;
+                    *fp++ = (double) P[pindex].birthMass;
+                    *fp++ = (double) P[pindex].aliveMass;
+                    *fp++ = (double) P[pindex].stochBirthMass;
+                    *fp++ = (double) P[pindex].stochAliveMass;
+                    *fp++ = (double) P[pindex].stochRemnantMass;
+                    *fp++ = (double) P[pindex].nonStochBirthMass;
+                    *fp++ = (double) P[pindex].nonStochAliveMass;
+                    *fp++ = (double) P[pindex].nonStochRemnantMass;
+                    *fp++ = (double) P[pindex].stellarMass;
+                    *fp++ = (double) P[pindex].stochStellarMass;
+                    *fp++ = (double) P[pindex].nonStochStellarMass;
+                    *fp++ = (double) P[pindex].formationTime;
+                    *fp++ = (double) P[pindex].curTime;
+                    *fp++ = (double) P[pindex].clusterAge;
+                    *fp++ = (double) P[pindex].lifetime;
+                    *fp++ = (double) P[pindex].stellarDeathMass;
+                    *fp++ = (double) P[pindex].A_V;
+                    *fp++ = (double) P[pindex].A_Vneb;
+                    *fp++ = (double) P[pindex].Lbol;
+                    *fp++ = (double) P[pindex].Lbol_ext;
+                    *fp++ = (double) P[pindex].tot_sn;
+                    *fp++ = (double) P[pindex].last_yield_time;
+                    n++;
+                }
+            break;
+
+        case IO_VGRADNORM: /* Velocity gradient */
+            for(n = 0; n < pc; pindex++)
+                if(P[pindex].Type == type)
+                {
+                    double vgn = 0.;
+                    for(k = 0; k < 3; k++)
+                    {
+                        for(int j = 0; j < 3; j++) 
+                          vgn += SphP[pindex].Gradients.Velocity[j][k]*SphP[pindex].Gradients.Velocity[j][k];
+                    }
+                    SphP[pindex].GradVelNorm = sqrt(vgn);

```

Comment:
If this variable is only used here, you could use a temporary local variable instead. But either one is fine, it's up to you.